libs/wingsdk/src/simulator/simulator.ts

```diff
@@ -1062,9 +1062,18 @@ export class Simulator {
     oldConfig: BaseResourceSchema,
     newConfig: BaseResourceSchema
   ) {
+    const state = (r: BaseResourceSchema) =>
+      JSON.stringify({
+        props: r.props,
+        type: r.type,
+        policyStatements: r.policy,
+      });
+
+    const invalidated = state(oldConfig) !== state(newConfig);
+
     // consult the resource's "plan()" method if it has one
     const instance = this.tryGetResource(path) as ISimulatorResourceInstance;
-    const plan = instance ? await instance.plan(newConfig) : UpdatePlan.AUTO;
+    const plan = instance ? await instance.plan(invalidated) : UpdatePlan.AUTO;
```

Comment:
Right now I'd prefer to not expose the oldConfig or newConfig since these values would contain tokens / unresolved values due to how the current update planner works.

I could be wrong, but my hunch if this plan() method is eventually going to be exposed in userland, then we want to avoiding leaking tokens to userland inflight code. From the mental model of a Winglang user, all preflight tokens should be resolved by the time they're used inflight.

I also couldn't think of any use cases for these fields (but if you need to add 'em later for some feature you want to implement, feel free to change the API of course...)